### DISCUSSION TITLE
"Tasks" im sae laufen voll. Ist das mein Problem?

### DISCUSSION THREAD

**Discussion by JoernBr** [camueller/SmartApplianceEnabler#618] [Q&A]
Hi,
Mein Problem:
mir ist beim Betrieb des sae die letzten Monate aufgefallen, daß dieser nach einigen Wochen "aussteigt" und nichts mehr schaltet. Bedarfsanforderungen fallen unter den Tisch und die Geräte laufen einfach nicht wie angefordert.

Meine Konfiguration:
Ich habe einen Raspberry 3B laufen, in der sae-Version 2.3.1 mit acht Geräten die der sae schalten kann.

Mein Vorgehen bei Auftreten des Fehlers:
Nach Eingabe des Befehles "sudo systemctl status smartapplianceenabler.service" kommt die Übersichtsanzeige:
![sae](https://github.com/user-attachments/assets/4159f65d-3d98-4366-a4c5-ebe7954d3b3d)
Um dieses Problem zu beheben, stoppe ich den sae-Dienst mit "sudo systemctl stop smartapplianceenabler.service", und starte ihn direkt wieder mit dem "sudo systemctl start smartapplianceenabler.service". Danach ist die Anzeige bei Tasks wieder im niedrigen dreistellingen Bereich (aktuell 137) und alles geht wieder.

Meine Frage:
Was mir da aufgefallen ist, ist die Anzeige der "Tasks". Aktuell sind es 1593, das Limit steht bei 1595. Das verstehe ich so, daß da ein Zähler die Tasks (=Aufgaben?) mitzählt, und der Zähler ist nun an seiner oberen Grenze. Reagiert der sae deshalb nicht mehr auf Bedarfsanforderungen?
Kann mir jemand sagen was da falsch läuft und wie ich das nervige Starten und Stoppen vermeiden kann? Ca alle vier Wochen ist das nötig, besonders schweirig  ist es, wenn ich nicht zuhause bin und Muttis Waschmaschine nicht so läuft, wie sie es möchte 🙄.

MfG Jörn

**Comment by camueller** [comment]
Ich denke, das ich diesen Fehler vor einigen Monaten gefixt habe. Ich muss mal wieder ein Release bauen. Kannst Du testen, ob mit dieser Test-Version von 2.3.1 das Problem behoben ist:  https://drive.google.com/file/d/1NTPOWaoLXTx6791zPMfi_D638JVB87re/view?usp=drivesdk

**Reply by JoernBr** [reply]
Danke, habs draufgeladen und gestartet. Ich werde das mal die nächsten Wochen beobachten 👍

**Comment by JoernBr** [comment]
@camueller:  Hallo Axel, auch mit der Test-Version von 2.3.1 glaube ich nach Stand heute sagen zu können das das Volllaufen der Tasks immer noch stattfindet. Aktuell steht da wieder ein Wert von 1138 drin. Magst du dich dieses Themas nochmal annehmen?

<img width="503" alt="2024-09-08 10_54_24-" src="https://github.com/user-attachments/assets/3019ded4-8cb4-49de-be94-526c3659c4a5">

**Comment by camueller** [comment]
Ich habe gerade mal auf einem meiner produktieven Raspis nachgesehen. Dort waren es nach 3 Monaten und 18 Tagen:
```
Tasks: 1906 (limit: 3933)
```

Nach einen Neustart des SAE sah es so aus:
```
Tasks: 201 (limit: 3933)
```
Muss ich mal untersuchen ...

Zumindest hat es bisher nicht zu Problemen geführt, auch wenn es natürlich nicht sein soll.

**Comment by camueller** [comment]
Ich denke, dass ich die Ursache gefunden habe. Habe es gefixt und werde es erstmal bei mir ausprobieren. Danach stelle ich Dir auch eine neue Test-Version zur Verfügung.

**Comment by camueller** [comment]
Hier ist eine neue Test-Version:
https://drive.google.com/file/d/18bcXBY12T1WCABvY_ZHF9i9FI8SlY98L/view?usp=sharing

**Reply by JoernBr** [reply]
Danke, hab ich soeben eingespielt und werde mich melden, sobald ich eine Aussage treffen kann.

**Reply by JoernBr** [reply]
Hi Axel, ich habe deine letzte Testversion nun seit 16 Tagen auf dem Raspberry laufen. Das Problem ist offensichtlich erledigt. Die Tasks stehen nun heute bei 247 (Limit 1595) und zählen nicht immer weiter hoch.
Gute Arbeit, Danke!

**Comment by Florian415** [comment]
Hallo Axel, 
könntest Du die Testversion bitte nochmals hochladen? Ich habe das gleiche Problem - vollgelaufen mit 1593 Tasks. 
Danke & Gruss, Florian

**Reply by camueller** [reply]
Hier der Link: https://drive.google.com/file/d/1UJWShHhpsouqM5wyvnrg2XJD7o7BIEC8/view?usp=drivesdk

**Comment by Florian415** [comment]
Danke Dir Axel für die superschnelle Antwort! 
Grüsse aus dem Nebel, Florian
